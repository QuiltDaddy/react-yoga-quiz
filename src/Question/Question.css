.pose {
  height: 225px;
  max-width: 270px;
}

.link {
  display:inline-block;
  margin: auto;
  width: 200px;
  background-color: gainsboro;
  text-decoration: none;
  font-family: 'Oswald', sans-serif;
  color: #24A159;
  border-radius: 8px;
  box-shadow: 0px;
}

.link:focus {
  outline: none;
}

.link:hover {
  background-color: #ff476f;
  -webkit-transition-duration: 0.8s; /* Safari */
  transition-duration: 0.8s;
  color: white;
  cursor: pointer;
}

form {
  height: 150px;
  width: 200px;
  margin: auto;
}

fieldset {
  border: 0;
}

ul {
  list-style-type: none;
}

label {
  display: flex;
  transition: all 0.3s ease-out;
  cursor: pointer;
}

label:hover, label:focus {
  background-color: #ff476f;
  color: white;
}

label:focus, input:focus {
  outline: none;
}

input[type="radio"] {
 margin-top: 6px;
 margin-left: 1px;
 cursor: pointer;
}
