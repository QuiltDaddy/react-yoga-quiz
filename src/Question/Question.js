import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import theQuiz from '../theQuiz';import './Question.css';

class Question extends Component {
  componentDidMount(){
    if (!this.props.testing) {
      ReactDOM.findDOMNode(this.refs['radio']).focus();
    }
  }

  render() {
    const question = this.props.question > 0 ? this.props.question : 1
    const {checkAnswer} = this.props;


    return (
      <>
        <img 
          className='pose'
          src={theQuiz[question].pose}
          alt={theQuiz[question].alt} />
        <br/>
        <form>
          <fieldset>
            <legend>Name the yoga pose:</legend>
            <ul>
              {theQuiz[question].answers.map((item, index) => 
                <li key={index}>
                  <label>
                    <input
                      type='radio'
                      name='answer'
                      value={index}
                      defaultChecked={index === 0}
                      ref={index === 0 ? 'radio' : null} />
                    {item}
                  </label>
                </li>
              )}
            </ul>
            <Link
              to={`/question/${question}/answer`}
              className='link'
              onClick={() => {
                checkAnswer(question, (
                  theQuiz[question].correctAnswer === 
                  parseInt(document.querySelector('input[name="answer"]:checked').value)
                ) ? 1 : 0)
              }}>
              Check Answer
            </Link>
          </fieldset>
        </form>
      </>
    );
  }
}

export default Question;
