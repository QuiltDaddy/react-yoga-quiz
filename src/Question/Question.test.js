import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Question from './Question';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Question />
    </BrowserRouter>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
const tree = renderer
  .create(
    <BrowserRouter>
      <Question testing={true} />
    </BrowserRouter>)
  .toJSON();
expect(tree).toMatchSnapshot();  
});
