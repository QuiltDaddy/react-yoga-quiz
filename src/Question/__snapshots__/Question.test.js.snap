// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <img
    alt="People standing with feet hips-width apart, a slight bend in their knees, hands by their sides and palms facing forward."
    className="pose"
    src="pose1.png"
  />,
  <br />,
  <form>
    <fieldset>
      <legend>
        Name the yoga pose:
      </legend>
      <ul>
        <li>
          <label>
            <input
              defaultChecked={true}
              name="answer"
              type="radio"
              value={0}
            />
            Sturdy Foothill Pose
          </label>
        </li>
        <li>
          <label>
            <input
              defaultChecked={false}
              name="answer"
              type="radio"
              value={1}
            />
            Mountain Pose
          </label>
        </li>
        <li>
          <label>
            <input
              defaultChecked={false}
              name="answer"
              type="radio"
              value={2}
            />
            Upright Forward Pose
          </label>
        </li>
        <li>
          <label>
            <input
              defaultChecked={false}
              name="answer"
              type="radio"
              value={3}
            />
            Standing Corpse Pose
          </label>
        </li>
      </ul>
      <a
        className="link"
        href="/question/1/answer"
        onClick={[Function]}
      >
        Check Answer
      </a>
    </fieldset>
  </form>,
]
`;
